### ISSUE TITLE
[virtual-scroll]: collectionViewer.viewChange is emitted only after the observable returned from DataSource#connect has been emitted once

### ISSUE THREAD

**Issue by kimamula** [angular/material2#14589]
#### What is the expected behavior?

The following implementation of `DataSource#connect` should work.

```ts
connect(collectionViewer: CollectionViewer): Observable<ProjectListItem[]> {
  return collectionViewer.viewChange.pipe(
    switchMap(({ start, end }) => this.fetchData(start, end))
  );
}
```

#### What is the current behavior?

The above code does not work while the following works.

```ts
connect(collectionViewer: CollectionViewer): Observable<ProjectListItem[]> {
  return collectionViewer.viewChange.pipe(
    startWith({ start: 0, end: 0 }), // add this line
    switchMap(({ start, end }) => this.fetchData(start, end))
  );
}
```

IMO this behavior is really confusing.

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-nzrgbv?file=app%2Fcdk-virtual-scroll-data-source-example.ts

**Comment by jelbourn** [comment]
@kimamula can you clarify why you believe it should work that way?

**Comment by kimamula** [comment]
```ts
interface CollectionViewer {
  viewChange: Observable<{ start: number; end: number; }>;
}

connect(collectionViewer: CollectionViewer): Observable<T[] | ReadonlyArray<T>>;
```

This `connect()` interface tells one that it should "return new observable by observing the observable passed as the argument".
However, it is hard to imagine that in fact the returning observable locates more upstream than the passed observable in the initial data flow.
The situation is completely opposite to that many people would naturally imagine by looking the interface.

**Comment by kimamula** [comment]
`connect()` is required to return two kinds of data.

1. Initial data
2. Subsequent data in response to scrolling

To clarify its requirements, its interface should have been something like

```ts
connect(): {
  scrollObserver(collectionViewer: CollectionViewer): Observable<T[] | ReadonlyArray<T>>;
  initialData: Observable<T[] | ReadonlyArray<T>>;
};
```

or

```ts
connect(collectionViewer: CollectionViewer): Observable<T[] | ReadonlyArray<T>>;
getInitialData(): : Observable<T[] | ReadonlyArray<T>>;
```

**Comment by kimamula** [comment]
You don't need `startWith()` when you are using `DataSource` with `mat-table`.

https://stackblitz.com/edit/angular-7ixetg?file=app%2Ftable-basic-example.ts

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
